src/spikeinterface/core/waveform_tools.py

```diff
@@ -397,6 +398,280 @@ def _waveform_extractor_chunk(segment_index, start_frame, end_frame, worker_ctx)
                     wfs[pos, :, :] = wf[:, sparsity_mask[unit_ind]]
 
 
+def extract_waveforms_to_single_buffer(
+    recording,
+    spikes,
+    unit_ids,
+    nbefore,
+    nafter,
+    mode="memmap",
+    return_scaled=False,
+    file_path=None,
+    dtype=None,
+    sparsity_mask=None,
+    copy=False,
+    job_name=None,
+    **job_kwargs,
+):
+    """
+    Allocate a single buffer (memmap or or shared memory) and then distribute every waveform into it.
+
+    Contrary to extract_waveforms_to_buffers() all waveforms are extracted in the same buffer, so the spike vector is
+    needed to recover waveforms unit by unit. Importantly in case of sparsity, the channels are not aligned across
+    units.
+
+    Important note: for the "shared_memory" mode wf_array_info contains reference to
+    the shared memmory buffer, this variable must be reference as long as arrays as used.
+    And this variable is also returned.
+    To avoid this a copy to non shared memmory can be perform at the end.
+
+    Parameters
+    ----------
+    recording: recording
+        The recording object
+    spikes: 1d numpy array with several fields
+        Spikes handled as a unique vector.
+        This vector can be obtained with: `spikes = Sorting.to_spike_vector()`
+    unit_ids: list ot numpy
+        List of unit_ids
+    nbefore: int
+        N samples before spike
+    nafter: int
+        N samples after spike
+    mode: str
+        Mode to use ('memmap' | 'shared_memory')
+    return_scaled: bool
+        Scale traces before exporting to buffer or not.
+    file_path: str or path
+        In case of memmap mode, file to save npy file.
+    dtype: numpy.dtype
```

Comment:
To me it makes sense that the buffer array inhertings the type of the recording. Allowing this flexbility without testing seems unecessary to me and increasing the possbility of bugs.  Am I missing something?

Reply:
Actually I am thinking in how this interacts with return_scaled...

Reply:
Yes when return_scale=True the user have choice between float32 or float64 this is why we need this dtype.

Reply:
OK,

Reply:
I think it should just return what the recorder returns by default. Do you think that would be wrong?

Reply:
The logic of dtype is in waveform extactor. I think it is good to keep the logic there.
Here
https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/core/waveform_extractor.py#L1353

Reply:
Thanks for answering, yes, those three last points are outside of the scope of the PR.